x-pack/plugins/security_solution/public/detections/components/alerts_table/index.tsx

```diff
@@ -290,7 +293,11 @@ export const AlertsTableComponent: FC<DetectionEngineAlertTableProps> = ({
         showColumnSelector: !isEventRenderedView,
         showSortSelector: !isEventRenderedView,
       },
-      dynamicRowHeight: isEventRenderedView,
+      // if records are too less, we don't want table to be of fixed height.
+      // it should shrink to the content height.
+      // Height setting enables/disables virtualization depending on fixed/undefined height values respectively.
+      height: count >= 10 ? `${DEFAULT_DATA_GRID_HEIGHT}px` : undefined,
```

Comment:
@PhilippeOberti , this controls how `auto` vs `max` height of the table is managed.  I have below suggestion. Do you think below could work.

```suggestion
      height: count >= 20 ? `${DEFAULT_DATA_GRID_HEIGHT}px` : undefined,
```

Reply:
hey @logeekal I'm so sorry to get back on you so late. I understand a bit more now, thanks for the explanation!

I think I found what's bothering me the most:
 In our current setup, the pagination is at the very bottom of the table, right below the last item, even when the table renders less items that the number of items per page.
In the new system, there will very frequently be a white gap between the items in the table and the pagination, which looks really odd.
Discover - while having a similar behavior - is looking much better (to my subjective eyes) because their pagination is fixed at the bottom of the pate (always visible and not impacted by the user scrolling).

I understand that doing is exactly like Discover might be more challenging, and would be out of scope for this PR anyway as it's not really performance optimization but more look and feel...

I did run some tests for different resolution (focusing on the height of the window only). I tried to see what could be the table height to have the pagination at the very bottom of the page, right above the timeline banner at the very bottom.
Here are the numbers that I found:

| Resolution (height only) | Table height - KPI collapsed | Table height - KPI expanded |
| ------------- | ------------- | ------------- |
| 1080px  | 610px  | 300px  |
| 1200px  | 730px  | 420px  |
| 1440px  | 970px  | 660px  |

I'm not sure what to do here... We definitely don't want to handle KPI collapsed vs expanded, and I don't think it's really worth handling different screen sizes...

I'm not sure what to do at this time 😆

Reply:
Hey @PhilippeOberti , thanks for your analysis and feedback. I like it. I have added the changes we discussed in our meeting.

Reply:
thanks @logeekal! To summarize the discussion here:
- we're displaying a default height of 600px which works on most of the screens
- on 4K screens the table isn't taking the entire height available and we might in the future increase the 600px, but as this a less likely scenario to happen we can revisit this later.
- when the item count is lower than 20, the table height switches to automatic mode to avoid having empty space between the table items and the pagination
- now that we have virtualization, we can increase the default number of items per page. The decision is to go from 20 to 50 (Discover has a default of 100)